hardware_interface/source/skytraq_binary.c

```diff
@@ -267,9 +269,10 @@ uint8_t calc_checksum(uint8_t *message, uint16_t payload_length) {
  * @return false checksum failed
  */
 bool skytraq_verify_checksum(uint8_t *message) {
-    int checksum_location = strlen((char *)message) - 3; // 3rd to last byte is checksum
-    uint8_t expected = message[checksum_location];
     uint16_t payload_length = (message[2] << 8) | message[3]; // extract 16 bit payload size
+    int checksum_location = 4 + payload_length;
```

Comment:
Ah okay I think we were having issues with the checksum too but I never got to it